smartseq2_pipeline/run_multi_samples/scripts/parse_picard.py

```diff
@@ -0,0 +1,77 @@
+from crimson import picard
+import json
+import pandas as pd
+import argparse
+import sys
+from collections import OrderedDict
+import os
+
+def parse_picard_to_table_by_sample(input_metrics_list,output_filename):
+    mets=OrderedDict()
+    with open(input_metrics_list) as f:
+        for line in f:
+            met_fn = line.strip('\n')
+            base = os.path.basename(met_fn)
+            [sid,ftype] = os.path.splitext(base)
+            parsed = picard.parse(met_fn)
+            class_name = parsed['metrics']['class']
+            if class_name == "picard.analysis.AlignmentSummaryMetrics":
+                met = parsed['metrics']['contents'][2]
+            else:
+                met = parsed['metrics']['contents']
+            mets.update(met)
+
+    tab = pd.DataFrame.from_records([mets])
+    tab.index=[sid]
```

Comment:
Should `sid` have a default value? It's only defined inside of the for-loop

Reply:
sid is prefix of input file, for merging by sample, all input files should have the same prefix and ideally, just need to do once.

Reply:
Ok, thanks for the clarification! I'm not sure if this would ever happen, but in the case that there's an empty input file, `sid` would be undefined at this point so some error checking could be useful here